src/Microsoft.TestPlatform.Build/Tasks/TestTaskUtils.cs

```diff
@@ -0,0 +1,222 @@
+        if (task.VSTestConsolePath != null && !task.VSTestConsolePath.ItemSpec.IsNullOrEmpty())
+        {
+            builder.AppendFileNameIfNotNull(task.VSTestConsolePath);
+        }
+        else
+        {
+            builder.AppendFileNameIfNotNull(vsTestAppName);
+        }
+
+        // TODO log arguments in task
+        if (!task.VSTestSetting.IsNullOrEmpty())
+        {
+            isRunSettingsEnabled = true;
+            builder.AppendSwitchIfNotNull("--settings:", task.VSTestSetting);
+        }
+
+        if (task.VSTestTestAdapterPath != null && task.VSTestTestAdapterPath.Any())
+        {
+            foreach (var arg in task.VSTestTestAdapterPath)
+            {
+                builder.AppendSwitchIfNotNull("--testAdapterPath:", arg);
+            }
+        }
+
+        builder.AppendSwitchIfNotNull("--framework:", task.VSTestFramework);
+
+        // vstest.console only support x86 and x64 for argument platform
+        if (!task.VSTestPlatform.IsNullOrEmpty() && !task.VSTestPlatform.Contains("AnyCPU"))
+        {
+            builder.AppendSwitchIfNotNull("--platform:", task.VSTestPlatform);
+        }
+
+        builder.AppendSwitchIfNotNull("--testCaseFilter:", task.VSTestTestCaseFilter);
+
+        if (task.VSTestLogger != null && task.VSTestLogger.Any())
+        {
+            foreach (var arg in task.VSTestLogger)
+            {
+                builder.AppendSwitchIfNotNull("--logger:", arg);
+
+                if ((arg != null) && arg.StartsWith("console", StringComparison.OrdinalIgnoreCase))
+                {
+                    isConsoleLoggerSpecifiedByUser = true;
+                }
+            }
+        }
+
+        builder.AppendSwitchIfNotNull("--resultsDirectory:", task.VSTestResultsDirectory);
+
+        if (task.VSTestListTests)
+        {
+            builder.AppendSwitch("--listTests");
+        }
+
+        builder.AppendSwitchIfNotNull("--diag:", task.VSTestDiag);
+
+        if (task.TestFileFullPath == null || task.TestFileFullPath.ItemSpec.IsNullOrEmpty())
+        {
+            task.Log.LogError(Resources.Resources.TestFilePathCannotBeEmptyOrNull);
+        }
+        else
+        {
+            builder.AppendFileNameIfNotNull(task.TestFileFullPath);
+        }
+
+        // Console logger was not specified by user, but verbosity was, hence add default console logger with verbosity as specified
+        if (!task.VSTestVerbosity.IsNullOrWhiteSpace() && !isConsoleLoggerSpecifiedByUser)
+        {
+            var normalTestLogging = new List<string>() { "n", "normal", "d", "detailed", "diag", "diagnostic" };
+            var quietTestLogging = new List<string>() { "q", "quiet" };
+
+            string vsTestVerbosity = "minimal";
+            if (normalTestLogging.Contains(task.VSTestVerbosity.ToLowerInvariant()))
+            {
+                vsTestVerbosity = "normal";
+            }
+            else if (quietTestLogging.Contains(task.VSTestVerbosity.ToLowerInvariant()))
```

Comment:
I would introduce a variable to avoid calling twice `ToLowerInvariant`.

```suggestion
            var taskVsTestVerbosity = task.VSTestVerbosity.ToLowerInvariant();
            if (normalTestLogging.Contains(taskVsTestVerbosity))
            {
                vsTestVerbosity = "normal";
            }
            else if (quietTestLogging.Contains(taskVsTestVerbosity))
```